File: problem2/regio14aFixed.cpp
#include <iostream>
#include <vector>
using namespace std;

int nth(int skip, int i) {
	if(i<skip) return i;
	return i+1;
}

bool check(int skip, int n, vector<int> vec) {
	int d = vec[nth(skip,1)]-vec[nth(skip,0)];
	for(int i=2; i< n; i++) {
		if(vec[nth(skip,i)]-vec[nth(skip, i-1)]!= d)
			return false;
	}
	return true;
}

int main(void) {
	int n;
	cin >> n;
	vector<int> vec(n+1);
	for (int i = 0; i < n + 1;  i++) {
		cin >> vec[i];
	}
	for(int skip=0; skip<n+1; skip++) {
		if(check(skip, n, vec)) {
			cout << vec[skip] << endl;
			return 0;
		}
	}
	return 0;
}

File: problem2/regio14aFixed.py
# デバッガ利用時など標準入力を使いたい場合は、変更を。
# ファイル名は環境によっては 'problem2/sample1.txt' などに変更を。
# input = open('sample1.txt', 'r').readline

def nth(skip, i):
    if i < skip:
        return i
    else:
        return i + 1


def check(skip, n, lst):
    d = lst[nth(skip, 1)] - lst[nth(skip, 0)]
    for i in range(2, n):
        di = lst[nth(skip, i)] - lst[nth(skip, i - 1)]
        if di != d:
            return False
    return True


def main():
    n = int(input())
    line = input()
    lst = list(map(int, line.split()))
    for skip in range(n + 1):
        if check(skip, n, lst):
            print(lst[skip])
            return


main()
